sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -171,9 +187,17 @@ class __SYCL_EXPORT modifiable_command_graph {
   template <typename T> node add(T CGF, const property_list &PropList = {}) {
     if (PropList.has_property<property::node::depends_on>()) {
       auto Deps = PropList.get_property<property::node::depends_on>();
-      return addImpl(CGF, Deps.get_dependencies());
+      node Node = addImpl(CGF, Deps.get_dependencies());
+      if (PropList.has_property<property::node::depends_on_all_leaves>()) {
+        addGraphLeafDependencies(Node);
+      }
+      return Node;
+    }
+    node Node = addImpl(CGF, {});
+    if (PropList.has_property<property::node::depends_on_all_leaves>()) {
+      addGraphLeafDependencies(Node);
```

Comment:
Should we move that call into `addImpl`? Looks like many/all uses need it.

Reply:
At the moment graph properties are handled at this level (graph front-end) and not in the graph implementation itself.
So, if you don't have a strong preference to see this call moved to addImpl, we probably prefer to keep it there.

Reply:
Looks like they all stay inside graph code and don't leak into regular SYCL RT code, so I'm fine.